### PR TITLE
Port e2e tests to separate repo

### PR DESCRIPTION
**Pull request by gidjin**
# SC-722

## Description 

Initial port of e2e tests to the portal repo. First step in moving the e2e tests out into a separate repo. To avoid cluttering the repo I created a folder called `e2e` to store all the e2e tests in.

## Review Notes

Also this should resolve SC-723 as it was easy enough to include in this PR.

Please review the github settings to ensure none were missed, though I copied it from our client repo.

## How to run the tests

Stop other running containers and remove all stopped containers

```sh
cd e2e
docker ps -qa | xargs docker rm -f
```

Start the system in docker. Note `--build` forces the building of the images, `--force-recreate` will remove the containers if they exist already, and `--remove-orphans` will remove any other running containers.

```sh
docker compose up --build --force-recreate --remove-orphans
```
Note: I've had a couple times where the CMS doesn't start up properly, I'm not clear why but ctrl-C and running the above command a second time works.

Once the above services are running you should run all the cypress & playwright tests. Everything should pass, though you might have to re-run things...

```sh
yarn
```

#### Cypress Based Tests

```sh
yarn cypress:install
yarn cypress:dev
```

#### Playwright Based Tests

These are still named `e2e` as I think the desire is to eventually migrate all our tests from cypress to playwright.

```sh
yarn e2e:install
yarn e2e:test
```

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  - name: 'needs: refinement'
+    color: '#ffffe5'
+
+branches:
+  - name: main
+    # Branch Protection settings. Set to null to disable
+    protection:
+      required_pull_request_reviews:
+        required_approving_review_count: 1
+        dismiss_stale_reviews: true
```
Comment (esacteksab) [inline .github/settings.yml:151]: I think we want this to false...and given I changed it in the UI, I think I owe both repos a PR to make this a permanent fix.

Comment (gidjin) [reply .github/settings.yml]: Good catch! I've updated the PR.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# Yarn Integrity file
+.yarn-integrity
+
+# dotenv environment variables file
+.env
+.envrc.*
+
+tsconfig.tsbuildinfo
+
+# asdf
```
Comment (esacteksab) [inline .gitignore:37]: I think you use `asdf` as do I. I'd love to include this if possible.

**Review by esacteksab — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (gidjin) [reply .gitignore]: Sure, I added `.tool-versions`

**Review by abbyoung — APPROVED** [review]
Followed your instructions with both portal and cms on `main` branch. After running cypress 2x, playwright 2x, all tests passed 🎉

**Review by jcbcapps — APPROVED** [review]

**Pull request merged**